packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts

```diff
@@ -141,14 +141,22 @@ export async function updateWalkthrough(context, lambdaToUpdate?: string) {
     .map(resource => resource.resourceName);
 
   if (lambdaFuncResourceNames.length === 0) {
-    context.print.error('No Lambda function resource to update. Use "amplify add function" to create a new function.');
-    return;
+    const errMessage = 'No Lambda function resource to update. Use "amplify add function" to create a new function.';
+    const error = new ResourceDoesNotExistError(errMessage);
+    error.name = 'FunctionResourceNotFound';
```

Comment:
Setting the error's name here and on line 156 is a little confusing. From what I can tell by scanning the codebase, it looks like we only overwrite the name on builtin `Error` objects.

Reply:
Ping @akshbhu